File: Core/src/main/java/dev/mrshawn/deathmessages/utils/EventUtils.java
package dev.mrshawn.deathmessages.utils;

import dev.mrshawn.deathmessages.DeathMessages;
import org.bukkit.event.Listener;

public class EventUtils {

	public static void registerEvents(Listener... listeners) {
		for (Listener listener : listeners) {
			DeathMessages.getInstance().getServer().getPluginManager().registerEvents(listener, DeathMessages.getInstance());
		}
	}
}

File: Core/src/main/kotlin/dev/mrshawn/deathmessages/kotlin/utils/EventUtils.kt
package dev.mrshawn.deathmessages.kotlin.utils

import dev.mrshawn.deathmessages.DeathMessages
import org.bukkit.Bukkit
import org.bukkit.event.Listener

object EventUtils {

	fun registerEvents(vararg listeners: Listener) {
		val pluginManager = Bukkit.getPluginManager()
		listeners.forEach { listener ->
			pluginManager.registerEvents(listener, DeathMessages.getInstance())
		}
	}

}